compiler/plc_diagnostics/src/diagnostics/error_codes/E032.md

```diff
@@ -1 +1,17 @@
-# Invalid number of parameters
+# Invalid number of arguments
+
+An invalid number of arguments was passed to a POU. For example
+
+```
+FUNCTION foo
+    (* ... *)
+END_FUNCTION
+
+FUNCTION main : DINT
+    foo('bar'); // Error, foo isn't expecting any arguments
+END_FUNCTION
+```
+
+Note that for `FUNCTION`s the argument count must match with the declared parameters and can be bigger if a variadic
+parameter is present. For stateful POUs the argument count can be shorter than the declared parameters or be bigger
+**if** variadic parameters are declared. This is because `VAR_INPUT`, `VAR_OUTPUT` arguments are optional.
```

Comment:
Programs and Function blocks do not support variadics, so smaller is allowed but never bigger AFAIK

Reply:
Yeah that makes sense, thanks!